framework/source/class/qx/lang/Type.js

```diff
@@ -101,7 +101,9 @@ qx.Bootstrap.define("qx.lang.Type",
       // JavaScript null and a null returned from calling DOM.
       // e.q. by document.getElementById("ReturnedNull").
       return (
-        value !== null && (
+        value !== null &&
+        // Check if value is NaN
+        value === value && (
```

Comment:
According to [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) isNaN coerces to number and then checks if is NaN, 

Number.isNaN() is [not supported in ie](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN)

Equality checking seems to give the [less surprises ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN#Testing_against_NaN)